core/src/main/java/org/apache/calcite/sql/parser/SqlParserUtil.java

```diff
@@ -139,38 +139,45 @@ public static java.sql.Timestamp parseTimestamp(String s) {
     return java.sql.Timestamp.valueOf(s);
   }
 
-  public static SqlDateLiteral parseDateLiteral(String s, SqlParserPos pos) {
-    final String dateStr = parseString(s);
+  public static SqlDateLiteral parseDateLiteralWithoutQuote(String dateStr, SqlParserPos pos) {
     final Calendar cal =
         DateTimeUtils.parseDateFormat(dateStr, Format.get().date,
             DateTimeUtils.UTC_ZONE);
     if (cal == null) {
       throw SqlUtil.newContextException(pos,
-          RESOURCE.illegalLiteral("DATE", s,
+          RESOURCE.illegalLiteral("DATE", dateStr,
               RESOURCE.badFormat(DateTimeUtils.DATE_FORMAT_STRING).str()));
     }
     final DateString d = DateString.fromCalendarFields(cal);
     return SqlLiteral.createDate(d, pos);
   }
 
-  public static SqlTimeLiteral parseTimeLiteral(String s, SqlParserPos pos) {
+  public static SqlDateLiteral parseDateLiteral(String s, SqlParserPos pos) {
     final String dateStr = parseString(s);
+    return parseDateLiteralWithoutQuote(dateStr, pos);
+  }
+
+  public static SqlTimeLiteral parseTimeLiteralWithoutQuote(String dateStr, SqlParserPos pos) {
```

Comment:
Why have a parameter `dateStr` for a function that parses a time literal? That seems confusing. Should it be `timeStr`? Or just stick to `s`.

Reply:
Good point. Let me change it. I keep it due to it's named as that in previous code.